### PR TITLE
[16.0][MIG] web_widget_domain_editor_dialog: Migration to 16.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT44374

Added support for multilevel groupings

ping @chienandalu @pilarvargas-tecnativa

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
Crashing test case:

- Enter a domain in the widget
- Click on the records count to show the selection popup
- Select any record
- Click select
- Crash:

```
TypeError: Cannot read properties of undefined (reading 'component')
    at DomainEditorDialog.get_domain
/web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.esm.js:67
/web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.esm.js:30
```

edit: indeed, you can't use the record selection at all

**Comment by CarlosRoca13** [comment]
Sorry, it should already be solved

File: web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.esm.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                });
+                group_domain = _.without(group_domain, "&");
+                groups_domain.push(group_domain);
+            }
+        }
+        return Domain.or(groups_domain).toList();
+    }
+
+    get_domain(resIds) {
+        const listController = findChildren(
```
Comment (chienandalu) [inline web_widget_domain_editor_dialog/static/src/js/widget_domain_editor_dialog.esm.js]: Maybe `listController` is a bit missguiding :)
```suggestion
        const dynamicList = findChildren(
```

**Review by chienandalu — COMMENTED** [review]
Now it works nicely. Just a nitpick :)

**Comment by CarlosRoca13** [comment]
Changed! Thanks :)

**Review by chienandalu — APPROVED** [review]
:+1: Great!

**Review by pilarvargas-tecnativa — APPROVED** [review]
:+1: thanks

**Comment by CarlosRoca13** [comment]
ping @pedrobaeza

**Pull request merged**